src/Microsoft.ApiDesignGuidelines.Analyzers/Core/PassSystemUriObjectsInsteadOfStrings.cs

```diff
@@ -24,15 +29,191 @@ public abstract class PassSystemUriObjectsInsteadOfStringsAnalyzer : DiagnosticA
+                if (node == null || node.SyntaxTree == null)
+                {
+                    // this is not from user code. we are not interested.
+                    return;
+                }
+
+                var expressionNode = _expressionGetter(node);
+                if (expressionNode == null)
+                {
+                    // we don't have right expression node to check overloads
+                    return;
+                }
+
+                // REVIEW: why IOperation doesn't contain things like compilation and semantic model?
+                //         it seems wierd that I need to do this to get thsoe.
+                var model = _compilation.GetSemanticModel(node.SyntaxTree);
+
+                if (!CheckCandidate(model, method, expressionNode, context.CancellationToken))
+                {
+                    // not a method we are interested in.
+                    return;
+                }
+
+                context.ReportDiagnostic(expressionNode.CreateDiagnostic(Rule));
+            }
+
+            private bool CheckCandidate(SemanticModel model, IMethodSymbol method, SyntaxNode node, CancellationToken cancellationToken)
+            {
+                // due to limitation of using "this" in lambda in struct
+                var stringType = _string;
+                var stringParameters = method.Parameters.Where(p => p.Type?.Equals(stringType) == true);
+                if (!stringParameters.Any())
+                {
+                    // no string parameter. not interested.
+                    return false;
+                }
+
+                // now do cheap string check whether those string parameter contains uri word list we are looking for.
+                if (!CheckStringParametersQuick(stringParameters, cancellationToken))
+                {
+                    // no string parameter that contains what we are looking for.
+                    return false;
+                }
+
+                // now we make sure we actually have overloads that contains uri type parameter
+                if (!CheckOverloads(model, node, cancellationToken))
+                {
+                    // no overload that contains uri as parameter
+                    return false;
+                }
+
+                // now we do more expensive word parsing to see whether this is really right one to report it.
+                return CheckStringParametersSlow(stringParameters, cancellationToken);
+            }
+
+            private bool CheckOverloads(SemanticModel model, SyntaxNode node, CancellationToken cancellationToken)
+            {
+                var uriType = _uri;
+                foreach (var overload in model.GetMemberGroup(node, cancellationToken).OfType<IMethodSymbol>())
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    if (overload.Parameters.Any(p => p.Type?.Equals(uriType) == true))
```

Comment:
You should check that `overload != method`, the original method. See FXCop check for this: http://index/#UsageRules/PassSystemUriObjectsInsteadOfStrings.cs,67